### DISCUSSION TITLE
Define type for Result ?

### DISCUSSION THREAD

**Discussion by vivekascoder** [wevm/wagmi#988] [Q&A] [answered]
For example, I have this useContractRead call here.
```ts
  const symbolTx = useContractRead({
    addressOrName: parsedDao?.daoToken || "",
    contractInterface: CONFIG.INTERFACES.DAO_TOKEN.abi,
    functionName: "symbol",
  });
```

I can't specify the type of symbolTx.data or can I.

Wouldn't it be cool if `wagmi` uses something like `useContractReact<IResult>({});` so it's clear what'll it return?

**Comment by tmm** [comment, marked as answer]
The return type (as well as function names, arguments, etc.) will be automatically inferred based on your ABI once https://github.com/wagmi-dev/wagmi/pull/941 is complete.

**Comment by vivekascoder** [comment]
Just saw the demo from ETHOnline, looks cool. The library that you're mentioning is https://github.com/wagmi-dev/abitype?

**Reply by tmm** [reply]
That's the one!
